packages/amplify-graphql-auth-transformer/src/vtl-generator/ddb/resolvers/helpers.ts

```diff
@@ -165,12 +166,31 @@ export const iamExpression = (
         iff(not(ref(IS_AUTHORIZED_FLAG)), iamCheck(role.claim!, set(ref(IS_AUTHORIZED_FLAG), bool(true)), hasIdentityPoolId)),
       );
     });
-  } else {
+  }
+
+  if (genericIamAccessEnabled) {
```

Comment:
Is this the right place to evaluate genericIamAccess or do we need to do it before L159? Is `roles` populated for generic IAM cases? Should we only evaluate non-IAM cases if the IAM conditions don't apply, rather than evaluating every condition?

Reply:
I was reverse engineering this from test snapshots by looking at expressions there.
I also added a diagram with desired algorithm in PR description.

Whether this is right place or 159 - I intend to figure this out though new tests that I'm yet to add. For now this is the best effort.
One thing I checked so far is that when this is disabled all existing tests are passing.

Reply:
I ended up refactoring this to look similar to RDS, i.e. just wrapping old rules with if/else.